tests/Microsoft.Identity.Test.Unit/WwwAuthenticateParametersTests.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.Identity.Client;
+
+namespace Microsoft.Identity.Test.Unit
+{
+    [TestClass]
+    public class WwwAuthenticateParametersTests
+    {
+        const string ClientIdKey = "client_id";
+        const string ResourceIdKey = "resource_id";
+        const string ResourceKey = "resource";
+        const string GraphGuid = "00000003-0000-0000-c000-000000000000";
+        const string AuthorizationUriKey = "authorization_uri";
+        const string AuthorizationKey = "authorization";
+        const string AuthorityKey = "authority";
+        const string AuthorizationValue = "https://login.microsoftonline.com/common/oauth2/authorize";
+        const string Realm = "realm";
+        const string EncodedClaims = "eyJpZF90b2tlbiI6eyJhdXRoX3RpbWUiOnsiZXNzZW50aWFsIjp0cnVlfSwiYWNyIjp7InZhbHVlcyI6WyJ1cm46bWFjZTppbmNvbW1vbjppYXA6c2lsdmVyIl19fX0=";
+        const string DecodedClaims = "{\"id_token\":{\"auth_time\":{\"essential\":true},\"acr\":{\"values\":[\"urn:mace:incommon:iap:silver\"]}}}";
+        const string DecodedClaimsHeader = "{\\\"id_token\\\":{\\\"auth_time\\\":{\\\"essential\\\":true},\\\"acr\\\":{\\\"values\\\":[\\\"urn:mace:incommon:iap:silver\\\"]}}}";
+        const string SomeClaims = "some_claims";
+
+        [TestMethod]
+        [DataRow("client_id=00000003-0000-0000-c000-000000000000", "authorization_uri=\"https://login.microsoftonline.com/common/oauth2/authorize\"")]
+        [DataRow("client_id=00000003-0000-0000-c000-000000000000", "authorization=\"https://login.microsoftonline.com/common/oauth2/authorize\"")]
+        [DataRow("client_id=00000003-0000-0000-c000-000000000000", "authority=\"https://login.microsoftonline.com/common/\"")]
+        [DataRow("client_id=00000003-0000-0000-c000-000000000000", "authority=\"https://login.microsoftonline.com/common\"")]
+        [DataRow("resource_id=00000003-0000-0000-c000-000000000000", "authorization=\"https://login.microsoftonline.com/common/oauth2/authorize\"")]
+        [DataRow("resource=00000003-0000-0000-c000-000000000000", "authorization=\"https://login.microsoftonline.com/common/oauth2/authorize\"")]
+        public void CreateWwwAuthenticateResponse(string resource, string authorizationUri)
+        {
+            // Arrange
+            HttpResponseMessage httpResponse = new HttpResponseMessage((HttpStatusCode)401)
```

Comment:
>(HttpStatusCode)401 [](http://example.com/codeflow?start=71&length=19)

I believe there is a static on HttpStatusCode which is the 401?

Reply:
[https://docs.microsoft.com/en-us/uwp/api/windows.web.http.httpstatuscode?view=winrt-20348](https://docs.microsoft.com/en-us/uwp/api/windows.web.http.httpstatuscode?view=winrt-20348)

HttpStatusCode.Unauthorized